frontend/src/pages/admin/user-edit.js

```diff
@@ -19,337 +18,287 @@ import {
+        .then(data => {
+          setFormName(data.name || '');
+          setFormEmail(data.email || '');
+          setFormActive(data.is_active);
+          setFormAdmin(data.is_superadmin);
+          setFormProjects(data.projects?.map(projectToOption) || []);
+          setUserLoaded(true);
+        })
+        .catch(error => console.error(error));
+    }
+  }, [params.id]);
 
-  getProjects () {
+  useEffect(() => {
+    // fetch projects once
     HttpClient.get([Settings.serverUrl, 'admin', 'project'])
-      .then(response => {
-        response = HttpClient.handleResponse(response, 'response');
-        return response.json();
+      .then(response => HttpClient.handleResponse(response))
+      .then(data => {
+        setProjects(data.projects);
       })
-      .then(data => this.setState({projects: data.projects, filteredProjects: data.projects}))
       .catch(error => console.error(error));
-  }
-
-  saveUser (userId, user) {
-    return HttpClient.put([Settings.serverUrl, 'admin', 'user', userId], {}, user)
-      .then(response => HttpClient.handleResponse(response, 'response'))
-      .then(response => response.json());
-  }
-
-  goBack = () => {
-    this.props.navigate(-1);
-  };
+  }, []);
 
-  onTextInputChange = (_event, value) => {
-    this.setState({inputValue: value});
-  };
-
-  componentDidMount () {
-    this.getUser(this.state.id);
-    this.getProjects();
-  }
-
-  componentDidUpdate (prevProps, prevState) {
-    if (
-      prevState.inputValue !== this.state.inputValue
-    ) {
-      let newSelectOptions = this.state.projects;
-      if (this.state.inputValue) {
-        newSelectOptions = this.state.projects.filter(menuItem =>
-          String(menuItem.title).toLowerCase().includes(this.state.inputValue.toLowerCase())
-        );
-
-        if (newSelectOptions.length === 0) {
-          newSelectOptions = [{
-            isDisabled: true,
-            value: {},
-            title: `No results found for "${this.state.inputValue}"`,
-          }];
-        }
+  useEffect(() => {
+    // handle input value changing for project filter
+    let newProjectOptions = projects;
+    if (inputValue) {
+      newProjectOptions = projects?.filter(menuItem =>
+        String(menuItem.title).toLowerCase().includes(inputValue.toLowerCase())
+      );
 
-        if (!this.state.isProjectsOpen) {
-          this.setState({ isProjectsOpen: true });
-        }
+      if (newProjectOptions.length === 0) {
+        newProjectOptions = [{
+          isDisabled: true,
+          value: {},
+          title: `No results found for "${inputValue}"`,
+        }];
       }
 
-      this.setState({
-        filteredProjects: newSelectOptions,
-      });
+      if (!isProjectsOpen){
+        setIsProjectsOpen(true);
+      }
     }
-  }
+    setFilteredProjects(newProjectOptions);
 
-  render () {
-    const { projects, inputValue, filteredProjects, user, userProjects } = this.state;
+  }, [inputValue, projects, isProjectsOpen]);
```

Comment:
`isProjectsOpen` can be removed from dependencies

Reply:
Only if I remove forcing it open when there is an input value present. ;)

Reply:
@mshriver oh I see, some specifics because of typeahead, then ignore my comments